cmd/vic-machine/common/images.go

```diff
@@ -0,0 +1,192 @@
+// Copyright 2016 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	log "github.com/Sirupsen/logrus"
+
+	"github.com/urfave/cli"
+	"github.com/vmware/vic/pkg/errors"
+	"github.com/vmware/vic/pkg/trace"
+	"github.com/vmware/vic/pkg/version"
+)
+
+const (
+	ApplianceImageKey  = "core"
+	LinuxImageKey      = "linux"
+	ApplianceImageName = "appliance.iso"
+	LinuxImageName     = "bootstrap.iso"
+
+	// An ISO 9660 sector is normally 2 KiB long. Although the specification allows for alternative sector sizes, you will rarely find anything other than 2 KiB.
+	ISO9660SectorSize = 2048
+	ISOVolumeSector   = 0x10
+	PublisherOffset   = 318
+)
+
+var (
+	images = map[string][]string{
+		ApplianceImageKey: {ApplianceImageName},
+		LinuxImageKey:     {LinuxImageName},
+	}
+)
+
+type Images struct {
+	ApplianceISO string
+	BootstrapISO string
+	OSType       string
+}
+
+func (i *Images) ImageFlags() []cli.Flag {
+	return []cli.Flag{
+		cli.StringFlag{
+			Name:        "appliance-iso, ai",
+			Value:       "",
+			Usage:       "The appliance iso",
+			Destination: &i.ApplianceISO,
+		},
+		cli.StringFlag{
+			Name:        "bootstrap-iso, bi",
+			Value:       "",
+			Usage:       "The bootstrap iso",
+			Destination: &i.BootstrapISO,
+		},
+	}
+}
+
+func (i *Images) CheckImagesFiles(force bool) (map[string]string, error) {
+	defer trace.End(trace.Begin(""))
+
+	i.OSType = "linux"
+	// detect images files
+	osImgs, ok := images[i.OSType]
+	if !ok {
+		return nil, fmt.Errorf("Specified OS \"%s\" is not known to this installer", i.OSType)
+	}
+
+	imgs := make(map[string]string)
+	result := make(map[string]string)
+	if i.ApplianceISO == "" {
+		i.ApplianceISO = images[ApplianceImageKey][0]
+	}
+	imgs[ApplianceImageName] = i.ApplianceISO
+
+	if i.BootstrapISO == "" {
+		i.BootstrapISO = osImgs[0]
+	}
+	imgs[LinuxImageName] = i.BootstrapISO
+
+	for name, img := range imgs {
+		_, err := os.Open(img)
```

Comment:
Can you not use `os.Stat` here, too? Does the unused filehandle get `Close()`d in this case?